python/test/unit/language/test_core.py

```diff
@@ -55,6 +55,10 @@ def promotion_numpy_2_0():
         np._set_promotion_state(state)
 
 
+# No need to emulate NumPy 2.0 if the user has NumPy 2.0
+if np.__version__[0] != "1":
```

Comment:
Might be a bit simpler:
```suggestion
if hasattr(np, "_set_promotion_state"):
```

Reply:
It's just a stylistic issue I guess, but it's just that the context manager is enabling the NumPy 2.0 promotion semantics. Once we have NumPy 2.0 we don't need to do any of these, whether they have that utility or not.